sesman/chansrv/smartcard_pcsc.c

```diff
@@ -1686,18 +1689,30 @@ scard_process_cmd_wait_reader_state_change(struct trans *con,
                                            struct stream *in_s)
 {
     int rv;
-    //struct stream *out_s;
-    int timeOut;
+    int timeOut __attribute__((unused));
+    int reader_state_bytes;
+    struct stream *out_s;
     struct pcsc_uds_client *uds_client;
 
     LOG_DEVEL(LOG_LEVEL_DEBUG, "scard_process_cmd_wait_reader_state_change:");
-    in_uint32_le(in_s, timeOut);
-    LOG_DEVEL(LOG_LEVEL_DEBUG, "scard_process_cmd_wait_reader_state_change: timeOut %d",
-           timeOut);
-    //out_s = con->out_s;
     uds_client = (struct pcsc_uds_client *) (con->callback_data);
     uds_client->waiting = 1;
-    rv = 0;
+    if ((uds_client->vMajor == 4) && (uds_client->vMinor > 2)) {
```

Comment:
Presumably this will also be necessary for UDS V5, etc(?)

There's a couple of ways to fix this. You could store the version as major * 1000 + minor for example, which makes it easier to test.

Reply:
Thanks, modified